src/Core/Settings.h

```diff
@@ -642,6 +643,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(Bool, multiple_joins_try_to_keep_original_names, false, "Do not add aliases to top level expression list on multiple joins rewrite", 0) \
     M(Bool, optimize_distinct_in_order, true, "Enable DISTINCT optimization if some columns in DISTINCT form a prefix of sorting. For example, prefix of sorting key in merge tree or ORDER BY statement", 0) \
     M(Bool, optimize_sorting_by_input_stream_properties, true, "Optimize sorting by sorting properties of input stream", 0) \
+    M(Bool, force_aggregation_in_order, false, "Force use of aggregation in order on remote nodes during distributed aggregation. PLEASE, NEVER CHANGE THIS SETTING VALUE MANUALLY!", IMPORTANT) \
```

Comment:
if the initiator will execute aggregation in order, we force all remote nodes to also use this algorithm by sending this setting. `IMPORTANT` means that if a remote node does not recognise the setting (because it is a CH instance of an old version), it will fail the whole query with an exception (this is why we ask user to enable the new merging algorithm only after upgrade is finished).